### PR TITLE
File Factory Standardization

### PR DESCRIPTION
**Pull request by Windchild292**
Following up on https://github.com/MegaMek/megamek/pull/2165, this relocates and standardizes the file factories.

This requires the MekHQ-side to merge.

### CODE REVIEW

File: megamek/src/megamek/common/util/fileUtils/ScaledImageFileFactory.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+            @Override
+            public Object getItem() throws Exception {
+                // Cache the image on first use.
+                if (image == null) {
+                    image = createZippedImage(itemEntry, zipFile);
+                }
+                // Return a copy of the image.
+                return ImageUtil.getScaledImage(image, getWidth(), getHeight());
+            }
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/util/fileUtils/ScaledImageFileFactory.java:132]: Should we be caching the scaled image as long as height/width don't change? Or do we scale them a lot?

**Review by sixlettervariables — APPROVED** [review]
One question but no blockers.

File: megamek/src/megamek/common/util/fileUtils/ScaledImageFileFactory.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/common/util/fileUtils/ScaledImageFileFactory.java:132]: Pre-existing... but good point. Will swap that over

Comment (Windchild292) [reply megamek/src/megamek/common/util/fileUtils/ScaledImageFileFactory.java:132]: It has to do with cloning the image. I'll check and see if that's required anywhere else.

Comment (Windchild292) [reply megamek/src/megamek/common/util/fileUtils/ScaledImageFileFactory.java:132]: From testing it does not appear to be required, and thus I've pushed that change.

**Review by sixlettervariables — APPROVED** [review]

**Pull request merged**